### DISCUSSION TITLE
parseSpec and selection

### DISCUSSION THREAD

**Discussion by ricksmit3000** [uwdata/mosaic#240] [Q&A] [answered]
I'm trying to use parseSpec and pass it an selection that's created outside of the spec, so something:

```js
const globalSelection = vg.Selection.intersect();
const specs = {
  plot: [
    {
      mark: "areaY",
      data: {
        from: "kpis",
        filterBy: "global",
      },
      x: "date",
      y: "visits",
      stroke: "steelblue"
    },
  ],
};

vg.parseSpec(
  {
    ...specs,
    params: {
      global: { select: "intersect" },
    },
  },
  {
    options: { params: [["global", globalSelection]] },
  }
);
```
when I try `filterBy: "$global` it seems to create a new collection, and when I use `filterBy: "global"` the `selection` in the `FilterGroup` is "global" (so not an actual instance of the `Selection` class)

Is that possible, or am I using mosaic wrong? For context, I'm trying to create a date filter that would update all my clients to filter the same date range.

**Comment by mingfang** [comment, marked as answer]
In your spec, `filterBy: "$global"` is correct.
But your params may not be.
Try removing `options` and use just `params` 

> and `const globalSelection = vg.Selection.intersect();` 
> needs `$` like `const $globalSelection = vg.Selection.intersect();`.
 
Correction: The above is not true, `$` is not needed.  see comment below.

Here is a working example.
https://github.com/mingfang/mosaic-svelte/blob/master/src/routes/mosaic/%2Bpage.svelte#L35

I believe this is the relevant code
https://github.com/uwdata/mosaic/blob/main/packages/vgplot/src/spec/parse-spec.js#L374

IMO 
I don't agree that we should be forced to use `$` for selections.
Also the `$` conflicts with Svelte's use of `$`.

**Reply by jheer** [reply]
Thanks @mingfang. Within YAML/JSON specs the `$` prefix is currently required as the parser uses that to easily identify references to params (which can even be otherwise undefined). While Svelte also use `$`, Mosaic's requirement here only applies to these external YAML/JSON files. You are not required to use a `$` prefix if writing JavaScript code directly.

**Reply by mingfang** [reply]
@jheer I was able to confirm that `$` is not needed in Javascript code. Thanks.

**Reply by ricksmit3000** [reply]
thanks for the example @mingfang, that's very succinct and exactly what I'm trying to do! Also good to know that the `$` is not necessary @jheer, thank you
